scipy/stats/_distn_infrastructure.py

```diff
@@ -1540,6 +1541,23 @@ def _penalized_nnlf(self, theta, x):
         n_log_scale = len(x) * log(scale)
         return self._nnlf_and_penalty(x, args) + n_log_scale
 
+    def _get_shape_info(self):
+        return self._shape_info()
+
+
+class ShapeInfo:
+    def __init__(self, shape_name, integrality=False, domain=(-np.inf, np.inf),
+                 inclusive=(True, True)):
```

Comment:
minor point: is `inclusive=(True, True)` a good default value? It implies that there is some limiting distribution if the shape parameter is +/- inf

Reply:
Maybe. Not sure. The default is simply stated: the parameter is unrestricted. Another approach might be to figure out the most common set of restrictions on a shape and use that, but unless that accounts for most shapes, that might not be very good either. We could leave domain as it is and set `inclusive` to `(False, False)`, but there is not really a  practical difference between the two because the optimizers don't actually set a DV to infinity. I envisioned `inclusive` being used for shape parameters that must be, e.g., strictly positive.

Reply:
agree, there is no useful default value, this variable needs to be set carefully for each distribution. one could even say that domain and inclusive should not be optional.

``def __init__(self, shape_name, domain, inclusive,  integrality=False):...``

but i am also fine to keep it as it is

Reply:
OK. I marked these as private, so I'll let these be for now.